src/frequenz/sdk/timeseries/_quantities.py

```diff
@@ -0,0 +1,310 @@
+            raise TypeError(
+                "Expected exponent_unit_map to be a dict, "
+                f"got {type(exponent_unit_map)}"
+            )
+        if not 0 in exponent_unit_map:
+            raise ValueError("Expected a base unit for the type (for exponent 0)")
+        cls._exponent_unit_map = exponent_unit_map
+        super().__init_subclass__()
+
+    @property
+    def base_value(self) -> float:
+        """Return the value of this quantity in the base unit.
+
+        Returns:
+            The value of this quantity in the base unit.
+        """
+        return self._base_value
+
+    def isnan(self) -> bool:
+        """Return whether this quantity is NaN.
+
+        Returns:
+            Whether this quantity is NaN.
+        """
+        return math.isnan(self._base_value)
+
+    def isinf(self) -> bool:
+        """Return whether this quantity is infinite.
+
+        Returns:
+            Whether this quantity is infinite.
+        """
+        return math.isinf(self._base_value)
+
+    def __repr__(self) -> str:
+        """Return a representation of this quantity.
+
+        Returns:
+            A representation of this quantity.
+        """
+        return f"{type(self).__name__}(value={self._base_value}, exponent=0)"
+
+    def __str__(self) -> str:
+        """Return a string representation of this quantity.
+
+        Returns:
+            A string representation of this quantity.
+        """
+        return self.__format__("")
+
+    def __format__(self, __format_spec: str) -> str:
+        """Return a formatted string representation of this quantity.
+
+        If specified, must be of this form: `[0].{precision}`.  If a 0 is not given, the
+        trailing zeros will be omitted.  If no precision is given, the default is 3.
+
+        Example:
+            ```python
+            from frequenz.sdk.timeseries._quantities import Current
+            c = Current.from_amperes(1.2345)
+            assert f"{c:.2}" == "1.23 A"
+            c = Current.from_milliamperes(1.2345)
+            assert f"{c:.6}" == "1.2345 mA"
+            ```
+
+        Args:
+            __format_spec: The format specifier.
+
+        Returns:
+            A string representation of this quantity.
+
+        Raises:
+            ValueError: If the given format specifier is invalid.
+        """
+        keep_trailing_zeros = False
+        if __format_spec != "":
+            fspec_parts = __format_spec.split(".")
+            if (
+                len(fspec_parts) != 2
+                or fspec_parts[0] not in ("", "0")
+                or not fspec_parts[1].isdigit()
+            ):
+                raise ValueError(
+                    "Invalid format specifier. Must be empty or `[0].{precision}`"
+                )
+            if fspec_parts[0] == "0":
+                keep_trailing_zeros = True
+            precision = int(fspec_parts[1])
+        else:
+            precision = 3
```

Comment:
Add exception for types without `_exponent_unit_map`

Reply:
this was required based on an earlier version of the code.  You reviewed too quickly, before I could clean up the PR and mark it as ready for review. :-)

Reply:
mypy does check this.

Reply:
handled differently, with a default None value.